scripts/check-title-labels.lean

```diff
@@ -0,0 +1,274 @@
+    else if !type.endsWith ")" then
+      if type.back.isWhitespace then
+        errors := errors.push s!"error: the PR type should not end with a space"
+      else
+        errors := errors.push s!"error: the PR type should be of the form abbrev(scope)"
+    let (type, scope) := (splitAtPos type idx)
+    let scope := scope.dropRight 1
+    if !known_types.contains type then
+      errors := errors.push s!"error: the PR type should be one of {", ".intercalate (known_types.map (s!"\"{·}\""))}"
+    if scope.startsWith "Mathlib/" then
+      errors := errors.push s!"error: the PR scope must not start with 'Mathlib/'"
+    if scope.endsWith ".lean" then
+      errors := errors.push s!"error: a PR's scope must not end with '.lean'"
+    -- Disabling this for now, to reduce warning fatigue. Might enable this in the future.
+    -- if scope.contains '.' then
+    --  errors := errors.push s!"error: a PR's scope should be a directory, not a module"
+    -- Future: we could check if `scope` describes a directory that actually exist.
+  else
+    errors := errors.push "error: the PR type should be of the form abbrev or abbrev(scope)"
+  return errors
+
+
+/--
+Check if a combination of PR labels is considered contradictory.
+`labels` contains the names of this PR's labels.
+-/
+def hasContradictoryLabels (labels : Array String) : Bool := Id.run do
+  -- Combine common labels.
+  let substitutions := [
+    ("ready-to-merge", "bors"), ("auto-merge-after-CI", "bors"),
+    ("blocked-by-other-PR", "blocked"), ("blocked-by-core-PR", "blocked"),
+    ("blocked-by-batt-PR", "blocked"), ("blocked-by-qq-PR", "blocked"),
+  ]
+  let mut canonicalise : Std.HashMap String String := .ofList substitutions
+  let normalised_labels := labels.map fun label ↦ (canonicalise.getD label label)
+  -- Test for contradictory label combinations.
+  if normalised_labels.contains "awaiting-review-DONT-USE" then
+    return true
+  else if normalised_labels.contains "bors" &&
+      ["awaiting-CI", "awaiting-zulip", "WIP"].any normalised_labels.contains then
+    return true
+  else if ["delegated", "awaiting-zulip"].all normalised_labels.contains then
+    return true
+  else
+    return false
+
+
+open Cli in
+/-- Implementation of the `check-title-labels` command line program.
+The exit code is the number of violations found. -/
+def checkTitleLabelsCLI (args : Parsed) : IO UInt32 := do
+  let title := (args.positionalArg! "title").value
+  let labels : Array String := args.variableArgsAs! String
+  let mut numberErrors := 0
+  if !labels.contains "WIP" then
+    -- We do not complain about WIP PRs.
+    -- The title should be of the form "abbrev: main title" or "abbrev(scope): main title".
+    let titleErrors := validateTitle title
+    for err in titleErrors do
+      IO.println err
+      numberErrors := numberErrors + 1
+    -- TODO: can I use this, with some casting?
+    -- numberErrors := numberErrors + titleErrors.size
```

Comment:
you can use [Nat.toUInt32](https://leanprover-community.github.io/mathlib4_docs/Init/Data/UInt/BasicAux.html#Nat.toUInt32) (or maybe preferably [UInt32.ofNatTruncate](https://leanprover-community.github.io/mathlib4_docs/Init/Data/UInt/BasicAux.html#UInt32.ofNatTruncate)?) for that. This cast will likely never be implicit, since it changes data.